cas-server-support-oauth/src/main/java/org/jasig/cas/config/OAuthConfiguration.java

```diff
@@ -51,14 +73,35 @@
     public Config oauthSecConfig() {
         final CasClient oauthCasClient = new CasClient(this.casLoginUrl);
         oauthCasClient.setName(CAS_OAUTH_CLIENT);
+        oauthCasClient.setCallbackUrlResolver((url, context) -> {
```

Comment:
The /authorize endpoint is called with the redirect_uri parameter. Protected by pac4j, this url is saved and the redirection to the CAS server happens. Then, after a successful authentication, you're redirected back to the /callback url and finally to the originally requested url (something we should improve in the future: https://github.com/Jasig/cas/issues/1571).
If you take a look at the controller, you see that the client_id is retrieved and checked with redirect_uri (https://github.com/Jasig/cas/blob/master/cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20AuthorizeController.java#L164), then the real service is retrieved from the client_id: https://github.com/Jasig/cas/blob/master/cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20AuthorizeController.java#L61. It means we retrieve the appropriate CAS service (the one matching the client_id and redirect_uri) and thus get the right attributes release policy for example.

So I think everything works properly unless there is a bug I missed. I guess you should try without this specific callback url resolver and see that everything is working properly (especially attributes releases).